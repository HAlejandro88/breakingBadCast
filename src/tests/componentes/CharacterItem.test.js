import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CharacterItem from '../../components/characters/CharacterItem';



describe('pruebas en el componente <CharacterItem/>', () => {

    const data = {
        name: 'Batman',
        img: 'data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAkGBxMTEhUTExMVFRUXGBYXGRgXFxYWFxoYGBgXFhoYFxcZHighGBslHRUVITEiJSkrLi4uFx8zODMtNygtLisBCgoKDg0OGhAQGi0lHyUrLS0tLS0',
        portrayed: 'Alejandro Soto',
        nickname: 'Bruce wayne',
        birthday: 'september 6',
        status: 'active'
    }

    test('debe de comparase contra el snapshot', () => {
        const wrapper = shallow(<CharacterItem item={data} />);
        expect(wrapper).toMatchSnapshot();
    })


    test('debe de traer los siguiente elementos en el snapShot', () => {
        const wrapper = shallow(<CharacterItem item={data} />);
        expect(wrapper.find('img').prop('src')).toBe(data.img)
        expect(wrapper.find('h1').text().trim()).toBe(data.name);
    })

    test('de mostrar la lista con los items', () => {
        const wrapper = shallow(<CharacterItem item={data} />);
        const lista = wrapper.find('ul');
        const listaCompleta = lista.children('li');
        expect(listaCompleta.length).toBe(4)
    })

})