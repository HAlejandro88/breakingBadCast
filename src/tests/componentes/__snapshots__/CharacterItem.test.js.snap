// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <CharacterItem/> debe de comparase contra el snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-inner"
  >
    <div
      className="card-front"
    >
      <img
        alt=""
        src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAkGBxMTEhUTExMVFRUXGBYXGRgXFxYWFxoYGBgXFhoYFxcZHighGBslHRUVITEiJSkrLi4uFx8zODMtNygtLisBCgoKDg0OGhAQGi0lHyUrLS0tLS0"
      />
    </div>
    <div
      className="card-back"
    >
      <h1>
        Batman
      </h1>
      <ul>
        <li>
          <strong>
            Actor Name:
          </strong>
           
          Alejandro Soto
        </li>
        <li>
          <strong>
            Nickname:
          </strong>
           
          Bruce wayne
        </li>
        <li>
          <strong>
            Birthday:
          </strong>
           
          september 6
        </li>
        <li>
          <strong>
            Status:
          </strong>
           
          active
        </li>
      </ul>
    </div>
  </div>
</div>
`;
